### PR TITLE
Tips area

### PR DESCRIPTION
**Pull request by Vinnl**
~~(Still set to draft because [the feature-parity PRs](https://github.com/mozilla/fx-private-relay/issues?q=is%3Aopen+label%3Areact+draft%3Afalse) have higher priority. Additionally, this still lists tips that have been moved to the New Feature Announcement area, and do not have final content yet, to be able to demo the functionality.)~~

# New feature description

Adds a "Help & Tips" area that can give the user pointers at relevant moments.

# Screenshot (if applicable)

https://user-images.githubusercontent.com/4251/159662158-af185af8-9953-4cf2-bae2-35ae1c4a7746.mp4

# How to test

Log in with a user with a subdomain set, but no custom aliases yet. You should see at least a tip explaining how to create custom aliases. The title will be shown relatively large at first. You can then either minimise it, or hit "Learn more" to expand the full tip. After minimising, it should not be shown in the initial relatively large size anymore for that user.

# Checklist

- [ ] l10n dependencies have been merged, if any.
- [ ] All acceptance criteria are met.
- [x] I've added or updated relevant docs in the docs/ directory.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/static/scss/libs/protocol/css/includes/tokens/dist/index.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: frontend/src/components/dashboard/Tips.tsx [hunk 1]
```diff
@@ -1,128 +1,238 @@
...
+  const allAliases = (props.customAliases as AliasData[]).concat(
+    props.randomAliases
+  );
   if (
     getRuntimeConfig().featureFlags.criticalEmailsTip === true &&
-    !dismissals.criticalEmails.isDismissed
+    props.profile.has_premium &&
+    allAliases.length > 0
+    // TODO: uncomment this when the promo-blocking React UI is merged:
+    // && allAliases.every((alias) => !alias.block_list_emails
```
Comment (Vinnl) [inline frontend/src/components/dashboard/Tips.tsx:69]: Relevant PR: https://github.com/mozilla/fx-private-relay/pull/1673

Comment (Vinnl) [reply frontend/src/components/dashboard/Tips.tsx]: No longer relevant, as this PR doesn't include this tip anymore.

File: frontend/src/components/dashboard/Tips.tsx [hunk 2]
```diff
@@ -222,24 +256,26 @@ const PanelDot = (props: PanelDotProps) => {
...
+          <title>{alt}</title>
+          <circle
+            style={{
+              fill: "currentcolor",
+            }}
+            cx="4"
+            cy="4"
+            r="4"
+          />
+        </svg>
```
Comment (lloan) [inline frontend/src/components/dashboard/Tips.tsx:277]: question: Is this an icon we could potentially reuse in the future? Should this go in the `Icons.tsx` file?

Comment (lloan) [inline frontend/src/components/dashboard/Tips.tsx:277]: question: Is this an icon we could potentially reuse in the future? Should this go in the `Icons.tsx` file?

**Review by lloan — COMMENTED** [review]
@Vinnl I left a couple of questions. Mostly about modularizing `Tips.tsx`. Another question I have is, should this be visible on mobile? Or is it only for desktop? I tested this on mobile and this is what I see: 

image

File: frontend/src/components/dashboard/Tips.tsx [hunk 3]
```diff
@@ -1,128 +1,186 @@
...
   );
 };
 
+type GenericTipProps = {
+  title: string;
+  content: ReactNode;
+};
+// This component will probably be used for future tips that are yet to be added:
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const GenericTip = (props: GenericTipProps) => {
```
[the remark this answers is not in the document]
Comment (Vinnl) [reply frontend/src/components/dashboard/Tips.tsx]: We don't have a stance yet, but we could have one if you have a preference :) Elsewhere we've already done it, so for consistency I moved this one out now as well: 6d605973

(I also snuck in a small unrelated fix for a warning that came up in the console when running the website: e82eb2c9)

File: frontend/src/components/dashboard/Tips.module.scss [hunk 4]
```diff
@@ -65,14 +130,28 @@
         cursor: pointer;
         // Avoid layout shifts when adding a bottom border on focus:
         border: 2px solid transparent;
+        // border-radius: 100%;
```
[the remark this answers is not in the document]
Comment (Vinnl) [reply frontend/src/components/dashboard/Tips.module.scss]: We don't - looks like I didn't pay attention when submitting this PR! 48ca07fa

**Comment by Vinnl** [comment]
@lloan 

> Another question I have is, should this be visible on mobile? Or is it only for desktop?

Mobile too, looks like I'm just not testing on narrow enough viewports. Fixed in 8819f992.

image

(I also snuck in a small unrelated fix for a warning that came up in the console when running the website: https://github.com/mozilla/fx-private-relay/commit/e82eb2c9bf5e22903f163e7cc79db9162aa2fa00)

**Review by lloan — APPROVED** [review]
praise: Awesome job with this!

File: frontend/src/components/dashboard/Tips.tsx (hunk 2 above)
Comment (Vinnl) [reply frontend/src/components/dashboard/Tips.tsx]: That sounds good - if we do ever want to reuse it, who-ever implements that will probably look at this code, and come up with moving it to `Icons.tsx` themselves.

**Comment by Vinnl** [comment]
Tony updated the requirements to also show the custom alias tip to users without a subdomain. @lloan I think the change is small enough that I'm going to go ahead and merge it, but if you want to check it after the fact just in case there's something to change about it after all, the change is in ec39ecc90dac42232055b51c695fbfdba70cd6dc. (It also resulted in the tips area becoming visible to our a11y tests, hence two straightforward fixes in there as well.)

**Pull request merged**